In [3]:
import pandas as pd
import numpy as np

## TENDERS.CSV 2015-2023


In [4]:
pd.set_option('display.max_columns', 50)

In [6]:
tenders = pd.read_csv("../../../tender2015-2023.csv")
tenders.head()

/var/folders/pf/565wgnj55rl181qjmrmqq69m0000gn/T/ipykernel_93291/948714849.py:1: DtypeWarning: Columns (12,13,14,15,16,18,19,20,22,23,24,25,26) have mixed types. Specify dtype option on import or set low_memory=False.
  tenders = pd.read_csv("../../../tender2015-2023.csv")


ocid   
0      ocds-5wno2w-MCO-ML-GOP-44-2015-2423  \
1  ocds-5wno2w-COTO-GADMEA-012-2015-175195   
2       ocds-5wno2w-MCO-GMCM-028-2015-2689   
3     ocds-5wno2w-COTO-GAD-I-009-2015-2783   
4    ocds-5wno2w-COTO-GAD-I-010A-2015-2783   

                                          release_id   
0   MCO-ML-GOP-44-2015-2423-2021-12-03T00:34:30.539Z  \
1  COTO-GADMEA-012-2015-175195-2021-12-03T01:26:4...   
2    MCO-GMCM-028-2015-2689-2021-12-02T21:46:52.547Z   
3  COTO-GAD-I-009-2015-2783-2021-12-03T00:04:34.256Z   
4  COTO-GAD-I-010A-2015-2783-2021-12-02T19:49:54....   

                            id                        title   
0      MCO-ML-GOP-44-2015-2423      MCO-ML-GOP-44-2015-2423  \
1  COTO-GADMEA-012-2015-175195  COTO-GADMEA-012-2015-175195   
2       MCO-GMCM-028-2015-2689       MCO-GMCM-028-2015-2689   
3     COTO-GAD-I-009-2015-2783     COTO-GAD-I-009-2015-2783   
4    COTO-GAD-I-010A-2015-2783    COTO-GAD-I-010A-2015-2783   

                                     description    status   
0            AVIARIO PARA EL ZOOLÓGICO MUNICIPAL  complete  \
1          CIERRE TECNICO DEL BOTADERO DE BASURA  complete   
2  CONSTRUCCION DEL PUENTE SOBRE EL RIO QUEBRADA  complete   
3     CONSTRUCCIÓN DEL POLIDEPORTIVO LA CAROLINA  complete   
4               CONSTRUCCIÓN POLIDEPORTIVO CHOTA  complete   

            procuringEntity_id   
0    EC-RUC-1160000240001-2423  \
1  EC-RUC-0860000320001-175195   
2    EC-RUC-1460000290001-2689   
3    EC-RUC-1060000260001-2783   
4    EC-RUC-1060000260001-2783   

                                procuringEntity_name  value_amount   
0                                  MUNICIPIO DE LOJA      22991.00  \
1                  Gobierno Municipal de Eloy Alfaro     319708.00   
2                                MUNICIPIO DE MORONA     127120.12   
3  GOBIERNO AUTONOMO DESCENTRALIZADO DE SAN MIGUE...     324938.70   
4  GOBIERNO AUTONOMO DESCENTRALIZADO DE SAN MIGUE...     499928.65   

  value_currency procurementMethod procurementMethodDetails   
0            USD         selective            Menor Cuantía  \
1            USD              open               Cotización   
2            USD         selective            Menor Cuantía   
3            USD              open               Cotización   
4            USD              open               Cotización   

  mainProcurementCategory  awardCriteria     tenderPeriod_startDate   
0                   works  ratedCriteria  2015-11-16T16:00:00-05:00  \
1                   works  ratedCriteria  2015-12-07T13:00:00-05:00   
2                   works  ratedCriteria  2015-10-02T19:00:00-05:00   
3                   works  ratedCriteria  2015-06-09T19:00:00-05:00   
4                   works  ratedCriteria  2015-07-06T20:00:00-05:00   

        tenderPeriod_endDate tenderPeriod_maxExtentDate   
0  2015-11-24T11:00:00-05:00  2015-11-24T11:00:00-05:00  \
1  2015-12-23T10:00:00-05:00  2015-12-23T10:00:00-05:00   
2  2015-10-15T15:00:00-05:00  2015-10-15T15:00:00-05:00   
3  2015-06-22T16:00:00-05:00  2015-06-22T16:00:00-05:00   
4  2015-07-16T17:00:00-05:00  2015-07-16T17:00:00-05:00   

   tenderPeriod_durationInDays    enquiryPeriod_startDate   
0                          7.0  2015-11-16T16:00:00-05:00  \
1                         15.0  2015-12-07T13:00:00-05:00   
2                         12.0  2015-10-02T19:00:00-05:00   
3                         12.0  2015-06-09T19:00:00-05:00   
4                          9.0  2015-07-06T20:00:00-05:00   

       enquiryPeriod_endDate enquiryPeriod_maxExtentDate   
0  2015-11-18T17:00:00-05:00   2015-11-18T17:00:00-05:00  \
1  2015-12-09T13:00:00-05:00   2015-12-09T13:00:00-05:00   
2  2015-10-06T19:00:00-05:00   2015-10-06T19:00:00-05:00   
3  2015-06-12T16:00:00-05:00   2015-06-12T16:00:00-05:00   
4  2015-07-08T20:00:00-05:00   2015-07-08T20:00:00-05:00   

   enquiryPeriod_durationInDays hasEnquiries   
0                           2.0          NaN  \
1                           2.0          NaN   
2            

In [8]:
tenders['procurementMethodDetails'].unique().tolist()

['Menor Cuantía',
 'Cotización',
 'Repuestos o Accesorios',
 'Comunicación Social – Contratación Directa',
 'Concurso publico',
 'Contratacion directa',
 'Comunicación Social – Proceso de Selección',
 'Subasta Inversa Electrónica',
 'Lista corta',
 'Contratación de Seguros',
 'Contratos entre Entidades Públicas o sus subsidiarias',
 'Bienes y Servicios únicos',
 'Obra artística, científica o literaria',
 'Licitación de Seguros',
 'Concurso Público por Lista Corta Desierta',
 'Asesoría y Patrocinio Jurídico',
 'Contratación Directa por Terminación Unilateral',
 'Contrataciones con empresas públicas internacionales',
 'Asesoría y Patrocinio Jurídico – Cons. puntuales y específicas',
 'Licitación',
 'Transporte de correo interno o internacional',
 'Lista Corta por Contratación Directa Desierta',
 'Catálogo electrónico - Compra directa en el convenio SICM-354-2011, ADQUISICION DE REPERTORIO DE MEDICAMENTOS, Selección de proveedores para la provisión de medicamentos a publicarse en el reper

In [ ]:
for Method in  tenders['procurementMethodDetails']:
    

## CONTRACTS 2015-2023.CSV

In [3]:
contracts  = pd.read_csv("../../../contracts2015-2023.csv")
contracts.head()

/var/folders/pf/565wgnj55rl181qjmrmqq69m0000gn/T/ipykernel_71421/1881580846.py:1: DtypeWarning: Columns (7,8,12,13) have mixed types. Specify dtype option on import or set low_memory=False.
  contracts  = pd.read_csv("../../../contracts2015-2023.csv")


ocid   
0  ocds-5wno2w-MCBS-GA-1-025-2015-15414  \
1   ocds-5wno2w-RE-2662-ZNO-E-15-253178   
2    ocds-5wno2w-RE-APPB-002-2015-58563   
3   ocds-5wno2w-RE-APPB--003-2015-58563   
4   ocds-5wno2w-RE-APPB--004-2015-58563   

                                          release_id   
0  MCBS-GA-1-025-2015-15414-2021-12-02T18:26:13.655Z  \
1   RE-2662-ZNO-E-15-253178-2021-12-02T18:23:45.497Z   
2    RE-APPB-002-2015-58563-2021-12-02T17:26:57.665Z   
3   RE-APPB--003-2015-58563-2021-12-02T17:40:37.028Z   
4   RE-APPB--004-2015-58563-2021-12-02T17:41:10.737Z   

                          id                     awardID  title  description   
0  345414-MCBS-GA-1-025-2015  1306919-MCBS-GA-1-025-2015    NaN          NaN  \
1    583114-RE-2662-ZNO-E-15    1306274-RE-2662-ZNO-E-15    NaN          NaN   
2    338198-RE-APPB-002-2015    1209404-RE-APPB-002-2015    NaN          NaN   
3   341007-RE-APPB--003-2015   1233569-RE-APPB--003-2015    NaN          NaN   
4   340114-RE-APPB--004-2015   1252707-RE-APPB--004-2015    NaN          NaN   

       status   contractPeriod_startDate     contractPeriod_endDate   
0  terminated  2015-05-14T00:00:00-05:00  2015-05-15T00:00:00-05:00  \
1  terminated  2015-05-13T00:00:00-05:00  2015-05-15T00:00:00-05:00   
2  terminated  2015-04-01T00:00:00-05:00  2015-07-30T00:00:00-05:00   
3  terminated  2015-04-14T00:00:00-05:00  2015-10-11T00:00:00-05:00   
4  terminated  2015-04-14T00:00:00-05:00  2015-08-12T00:00:00-05:00   

   contractPeriod_maxExtentDate  contractPeriod_durationInDays        amount   
0                           NaN                            1.0  3.240000e+01  \
1                           NaN                            2.0  3.500000e+08   
2                           NaN                          120.0  3.200000e+03   
3                           NaN                          180.0  1.607140e+03   
4                           NaN                          120.0  1.428570e+03   

  currency                 dateSigned   año  
0      USD  2015-05-14T00:00:00-05:00  2015  
1      USD  2015-05-13T00:00:00-05:00  2015  
2      USD  2015-04-01T00:00:00-05:00  2015  
3      USD  2015-04-14T00:00:00-05:00  2015  
4      USD  2015-04-14T00:00:00-05:00  2015

## ANÁLISIS EXPLORATORIO DE DATOS

### Tender


In [9]:
tenders

ocid   
0            ocds-5wno2w-MCO-ML-GOP-44-2015-2423  \
1        ocds-5wno2w-COTO-GADMEA-012-2015-175195   
2             ocds-5wno2w-MCO-GMCM-028-2015-2689   
3           ocds-5wno2w-COTO-GAD-I-009-2015-2783   
4          ocds-5wno2w-COTO-GAD-I-010A-2015-2783   
...                                          ...   
2144905     ocds-5wno2w-CE-20230002365202-404551   
2144906     ocds-5wno2w-CE-20230002365203-404551   
2144907     ocds-5wno2w-CE-20230002365204-404551   
2144908     ocds-5wno2w-CE-20230002365205-404551   
2144909     ocds-5wno2w-SIE-HGOIA-052-2023-49010   

                                                release_id   
0         MCO-ML-GOP-44-2015-2423-2021-12-03T00:34:30.539Z  \
1        COTO-GADMEA-012-2015-175195-2021-12-03T01:26:4...   
2          MCO-GMCM-028-2015-2689-2021-12-02T21:46:52.547Z   
3        COTO-GAD-I-009-2015-2783-2021-12-03T00:04:34.256Z   
4        COTO-GAD-I-010A-2015-2783-2021-12-02T19:49:54....   
...                                                    ...   
2144905  CE-20230002365202-404551-2023-08-31T01:13:01.558Z   
2144906  CE-20230002365203-404551-2023-08-31T01:13:01.576Z   
2144907  CE-20230002365204-404551-2023-08-31T01:13:01.595Z   
2144908  CE-20230002365205-404551-2023-08-31T01:13:01.611Z   
2144909  SIE-HGOIA-052-2023-49010-2023-08-23T04:02:26.479Z   

                                  id                              title   
0            MCO-ML-GOP-44-2015-2423            MCO-ML-GOP-44-2015-2423  \
1        COTO-GADMEA-012-2015-175195        COTO-GADMEA-012-2015-175195   
2             MCO-GMCM-028-2015-2689             MCO-GMCM-028-2015-2689   
3           COTO-GAD-I-009-2015-2783           COTO-GAD-I-009-2015-2783   
4          COTO-GAD-I-010A-2015-2783          COTO-GAD-I-010A-2015-2783   
...                              ...                                ...   
2144905     CE-20230002365202-404551  Orden de compra CE-20230002365202   
2144906     CE-20230002365203-404551  Orden de compra CE-20230002365203   
2144907     CE-20230002365204-404551  Orden de compra CE-20230002365204   
2144908     CE-20230002365205-404551  Orden de compra CE-20230002365205   
2144909     SIE-HGOIA-052-2023-49010           SIE-HGOIA-052-2023-49010   

                                               description    status   
0                      AVIARIO PARA EL ZOOLÓGICO MUNICIPAL  complete  \
1                    CIERRE TECNICO DEL BOTADERO DE BASURA  complete   
2            CONSTRUCCION DEL PUENTE SOBRE EL RIO QUEBRADA  complete   
3               CONSTRUCCIÓN DEL POLIDEPORTIVO LA CAROLINA  complete   
4                         CONSTRUCCIÓN POLIDEPORTIVO CHOTA  complete   
...                                                    ...       ...   
2144905  Orden de compra para adquirir los siguientes p...  complete   
2144906  Orden de compra para adquirir los siguientes p...  complete   
2144907  Orden de compra para adquirir los siguientes p...  complete   
2144908  Orden de compra para adquirir los siguientes p...  complete   
2144909  ADQUISICIÓN DE JERINGA 1 ml, MOVIL, JERINGA 10...  complete   

                  procuringEntity_id   
0          EC-RUC-1160000240001-2423  \
1        EC-RUC-0860000320001-175195   
2          EC-RUC-1460000290001-2689   
3          EC-RUC-1060000260001-2783   
4          EC-RUC-1060000260001-2783   
...                              ...   
2144905  EC-RUC-1768164300001-404551   
2144906  EC-RUC-1768164300001-404551   
2144907  EC-RUC-1768164300001-404551   
2144908  EC-RUC-1768164300001-404551   
2144909   EC-RUC-1768034790001-49010   

                                      procuringEntity_name  value_amount   
0                                        MUNICIPIO DE LOJA      22991.00  \
1                        Gobierno Municipal de Eloy Alfaro     319708.00   
2                                      MUNICIPIO DE MORONA     127120.12   
3        GOBIERNO AUTONOMO DESCENTRALIZADO DE SAN MIGUE...     324938.70   
4        GOBIERNO AUTONOMO DESCENTRALIZADO DE S

In [10]:
tenders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2144910 entries, 0 to 2144909
Data columns (total 30 columns):
 #   Column                        Dtype  
---  ------                        -----  
 0   ocid                          object 
 1   release_id                    object 
 2   id                            object 
 3   title                         object 
 4   description                   object 
 5   status                        object 
 6   procuringEntity_id            object 
 7   procuringEntity_name          object 
 8   value_amount                  float64
 9   value_currency                object 
 10  procurementMethod             object 
 11  procurementMethodDetails      object 
 12  mainProcurementCategory       object 
 13  awardCriteria                 object 
 14  tenderPeriod_startDate        object 
 15  tenderPeriod_endDate          object 
 16  tenderPeriod_maxExtentDate    object 
 17  tenderPeriod_durationInDays   float64
 18  enquiryPeriod_startDat

In [11]:
tenders.describe()

value_amount  tenderPeriod_durationInDays   
count  1.973767e+06                149672.000000  \
mean   1.776225e+04                     8.602511   
std    4.207695e+05                     9.265918   
min    0.000000e+00                     1.000000   
25%    1.938000e+01                     4.000000   
50%    1.260000e+02                     6.000000   
75%    1.288000e+03                    12.000000   
max    1.979756e+08                   637.000000   

       enquiryPeriod_durationInDays  awardPeriod_durationInDays   
count                 270594.000000               144290.000000  \
mean                       3.454685                    4.595537   
std                        5.981712                   13.191321   
min                        1.000000                    1.000000   
25%                        1.000000                    2.000000   
50%                        2.000000                    4.000000   
75%                        4.000000                    6.000000   
max                      391.000000                 1099.000000   

       numberOfTenderers           año  
count      355913.000000  2.144910e+06  
mean            3.882640  2.018293e+03  
std            14.961743  2.346642e+00  
min             1.000000  2.015000e+03  
25%             1.000000  2.016000e+03  
50%             2.000000  2.018000e+03  
75%             4.000000  2.020000e+03  
max           598.000000  2.023000e+03

In [12]:
tenders.columns

Index(['ocid', 'release_id', 'id', 'title', 'description', 'status',
       'procuringEntity_id', 'procuringEntity_name', 'value_amount',
       'value_currency', 'procurementMethod', 'procurementMethodDetails',
       'mainProcurementCategory', 'awardCriteria', 'tenderPeriod_startDate',
       'tenderPeriod_endDate', 'tenderPeriod_maxExtentDate',
       'tenderPeriod_durationInDays', 'enquiryPeriod_startDate',
       'enquiryPeriod_endDate', 'enquiryPeriod_maxExtentDate',
       'enquiryPeriod_durationInDays', 'hasEnquiries', 'eligibilityCriteria',
       'awardPeriod_startDate', 'awardPeriod_endDate',
       'awardPeriod_maxExtentDate', 'awardPeriod_durationInDays',
       'numberOfTenderers', 'año'],
      dtype='object')

In [13]:
tenders.isnull( ).sum( )

ocid                                  0
release_id                            0
id                                    0
title                                 0
description                           2
status                                0
procuringEntity_id                   82
procuringEntity_name                 82
value_amount                     171143
value_currency                   171128
procurementMethod                     0
procurementMethodDetails              0
mainProcurementCategory         1787469
awardCriteria                   1787469
tenderPeriod_startDate          1787469
tenderPeriod_endDate            1987964
tenderPeriod_maxExtentDate      1987964
tenderPeriod_durationInDays     1995238
enquiryPeriod_startDate         1787469
enquiryPeriod_endDate           1864167
enquiryPeriod_maxExtentDate     1864167
enquiryPeriod_durationInDays    1874316
hasEnquiries                    1953940
eligibilityCriteria             1958597
awardPeriod_startDate           1973782


In [14]:
tenders.duplicated( ).sum()
# No hay lineas duplicadas

0

In [15]:
# Seleccionar solo las columnas numéricas
df_numeric = tenders.select_dtypes(include=['float64',"int64"])

# Calcular la matriz de correlación solo para las columnas numéricas
df_numeric.corr()

# Mostrar la matriz de correlación



value_amount  tenderPeriod_durationInDays   
value_amount                      1.000000                     0.082207  \
tenderPeriod_durationInDays       0.082207                     1.000000   
enquiryPeriod_durationInDays     -0.002774                     0.895552   
awardPeriod_durationInDays             NaN                          NaN   
numberOfTenderers                 0.005489                     0.061236   
año                               0.010541                     0.127529   

                              enquiryPeriod_durationInDays   
value_amount                                     -0.002774  \
tenderPeriod_durationInDays                       0.895552   
enquiryPeriod_durationInDays                      1.000000   
awardPeriod_durationInDays                        0.024030   
numberOfTenderers                                 0.011860   
año                                               0.021722   

                              awardPeriod_durationInDays  numberOfTenderers   
value_amount                                         NaN           0.005489  \
tenderPeriod_durationInDays                          NaN           0.061236   
enquiryPeriod_durationInDays                    0.024030           0.011860   
awardPeriod_durationInDays                      1.000000          -0.022544   
numberOfTenderers                              -0.022544           1.000000   
año                                             0.071016           0.076772   

                                   año  
value_amount                  0.010541  
tenderPeriod_durationInDays   0.127529  
enquiryPeriod_durationInDays  0.021722  
awardPeriod_durationInDays    0.071016  
numberOfTenderers             0.076772  
año                           1.000000

#### Entidades Proveedoras

Existen 4176 entidades proveedoras desde el 2015 - 2023

In [16]:
tenders["procuringEntity_name"].nunique()

4176

In [17]:
tenders["procuringEntity_name"].unique().tolist()

['MUNICIPIO DE LOJA',
 'Gobierno Municipal de Eloy Alfaro',
 'MUNICIPIO DE MORONA',
 'GOBIERNO AUTONOMO DESCENTRALIZADO DE SAN MIGUEL DE IBARRA',
 'FUERTE MILITAR BOLIVAR',
 'GOBIERNO AUTONOMO DESCENTRALIZADO MUNICIPAL DEL CANTON GUANO',
 'MUNICIPIO DE MEJIA',
 'Gobierno Municipal de Santa Lucia',
 'Empresa Pública de hidrocarburos del Ecuador EP PETROECUADOR',
 'AUTORIDAD PORTUARIA DE PUERTO BOLIVAR',
 'MUNICIPIO DE MONTECRISTI',
 'CORPORACION NACIONAL DE TELECOMUNICACIONES',
 'CONSEJO NACIONAL ELECTORAL DELEGACION PROVINCIAL DEL CARCHI',
 'GOBIERNO AUTONOMO DESCENTRALIZADO PARROQUIAL LUIS CORDERO',
 'EMPRESA PUBLICA METROPOLITANA DE ASEO',
 'GOBIERNO AUTONOMO DESCENTRALIZADO DE LA PARROQUIA EL AIRO',
 'JUNTA PARROQUIAL SAN PABLO DE USHPAYACU',
 'Junta Parroquial de Alaquez',
 'GOBIERNO AUTONOMO DESCENTRALIZADO PARROQUIAL DE YANGANA',
 'UNIDAD ADMINISTRADORA DEL FONSAT',
 'GOBIERNO AUTONOMO DESCENTRALIZADO PARROQUIAL RURAL SANTA TERESITA',
 'CENTRO DE GESTIÓN Y DESARROLLO SOCIAL DEL G

In [18]:
tenders[tenders['ocid']=="ocds-5wno2w-CE-20230002374213-30375"]

ocid   
2039642  ocds-5wno2w-CE-20230002374213-30375  \

                                               release_id   
2039642  CE-20230002374213-30375-2023-08-31T01:28:26.479Z  \

                              id                              title   
2039642  CE-20230002374213-30375  Orden de compra CE-20230002374213  \

                                               description    status   
2039642  Orden de compra para adquirir los siguientes p...  complete  \

                 procuringEntity_id   
2039642  EC-RUC-0968580510001-30375  \

                                      procuringEntity_name  value_amount   
2039642  GOBIERNO AUTONOMO DESCENTRALIZADO PROVINCIAL D...         63.59  \

        value_currency procurementMethod   
2039642            USD         selective  \

                                  procurementMethodDetails   
2039642  Catálogo electrónico - Mejor oferta en el conv...  \

        mainProcurementCategory awardCriteria tenderPeriod_startDate   
2039642                     NaN           NaN                    NaN  \

        tenderPeriod_endDate tenderPeriod_maxExtentDate   
2039642                  NaN                        NaN  \

         tenderPeriod_durationInDays enquiryPeriod_startDate   
2039642                          NaN                     NaN  \

        enquiryPeriod_endDate enquiryPeriod_maxExtentDate   
2039642                   NaN                         NaN  \

         enquiryPeriod_durationInDays hasEnquiries eligibilityCriteria   
2039642                           NaN          NaN                 NaN  \

        awardPeriod_startDate awardPeriod_endDate awardPeriod_maxExtentDate   
2039642                   NaN                 NaN                       NaN  \

         awardPeriod_durationInDays  numberOfTenderers   año  
2039642                         NaN                NaN  2023

### Status

In [5]:
tenders['status'].unique()

array(['complete', 'active'], dtype=object)

In [7]:
# ver de que años son las ofertas o tenders que siguen activas
active_tenders = tenders[tenders['status']=="active"]['año'].value_counts()
active_tenders = pd.DataFrame(active_tenders)
active_tenders

count
año        
2022  43483
2021   8693
2023   3140
2015    336
2016    206
2017    163
2020     98
2018     96
2019     79

### Contract


In [5]:
contracts.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2075108 entries, 0 to 2075107
Data columns (total 15 columns):
 #   Column                         Dtype  
---  ------                         -----  
 0   ocid                           object 
 1   release_id                     object 
 2   id                             object 
 3   awardID                        object 
 4   title                          float64
 5   description                    float64
 6   status                         object 
 7   contractPeriod_startDate       object 
 8   contractPeriod_endDate         object 
 9   contractPeriod_maxExtentDate   float64
 10  contractPeriod_durationInDays  float64
 11  amount                         float64
 12  currency                       object 
 13  dateSigned                     object 
 14  año                            int64  
dtypes: float64(5), int64(1), object(9)
memory usage: 237.5+ MB


In [4]:
contracts.isnull().sum()

ocid                                   0
release_id                             0
id                                     0
awardID                                0
title                            2075108
description                      2075108
status                                 0
contractPeriod_startDate         1787557
contractPeriod_endDate           1787557
contractPeriod_maxExtentDate     2075108
contractPeriod_durationInDays    1787862
amount                           1787562
currency                         1787555
dateSigned                       1787560
año                                    0
dtype: int64

In [6]:
contracts['amount'].describe()

count    2.875460e+05
mean     7.561466e+07
std      1.967591e+10
min      1.000000e-02
25%      9.050000e+03
50%      2.050000e+04
75%      5.390000e+04
max      8.978000e+12
Name: amount, dtype: float64

In [13]:
## numero de nulos
contracts['amount'].isnull().value_counts()

amount
True     1787562
False     287546
Name: count, dtype: int64

In [20]:
## ver si el status de  los contratos que tienen amount null
contracts_amount_null = contracts[contracts['amount'].isnull()]['status'].value_counts()
contracts_amount_null = pd.DataFrame(contracts_amount_null)
contracts_amount_null

count
status             
terminated  1587262
active       200300

De los 1787562 contratos que tienen amount null 1587262 estan terminados y 200300 estan active

In [19]:
#saber a que años pertenecen los contratos activos con amount null
año_contract_amount_null = contracts[(contracts['amount'].isnull()) & (contracts['status']=='active')]['año'].value_counts()
año_contract_amount_null = pd.DataFrame(año_contract_amount_null)
año_contract_amount_null

count
año        
2022  50604
2023  35226
2021  21668
2017  20602
2018  16628
2016  15996
2015  15775
2019  14964
2020   8837

In [ ]:
contracts['amount'].min(),contracts['amount'].max()

In [8]:
contracts[contracts['amount']==contracts['amount'].min()]
# Hay algunos contractos con valor minimo

ocid   
1031     ocds-5wno2w-CDC-CONGOPE-2015-009-69008  \
1035     ocds-5wno2w-CDC-CONGOPE-2015-014-69008   
1173     ocds-5wno2w-CDC-EASYNET-009-2015-18087   
2134      ocds-5wno2w-CDC-GADMIS-010-2015-24339   
2137      ocds-5wno2w-CDC-GADMIS-013-2015-24339   
...                                         ...   
1504511     ocds-5wno2w-RE-MIDUVI-002-2020-2672   
1746821   ocds-5wno2w-CTBS-SRIPCH-001-2021-2438   
1750619   ocds-5wno2w-RE-CS-ANDEC-001-21-467908   
1860459    ocds-5wno2w-FONAG-RE-009-2022-227909   
1922094   ocds-5wno2w-MCO-BALAG-003-2022-108230   

                                                release_id   
1031     CDC-CONGOPE-2015-009-69008-2021-12-02T19:54:58...  \
1035     CDC-CONGOPE-2015-014-69008-2021-12-02T22:35:20...   
1173     CDC-EASYNET-009-2015-18087-2021-12-02T23:36:11...   
2134     CDC-GADMIS-010-2015-24339-2021-12-02T19:57:15....   
2137     CDC-GADMIS-013-2015-24339-2021-12-02T20:11:34....   
...                                                    ...   
1504511   RE-MIDUVI-002-2020-2672-2021-12-01T00:13:12.577Z   
1746821  CTBS-SRIPCH-001-2021-2438-2022-12-16T15:50:58....   
1750619  RE-CS-ANDEC-001-21-467908-2022-04-08T11:13:49....   
1860459  FONAG-RE-009-2022-227909-2022-10-13T15:36:27.1...   
1922094  MCO-BALAG-003-2022-108230-2023-01-13T09:58:37....   

                                   id                       awardID  title   
1031      392133-CDC-CONGOPE-2015-009  1462887-CDC-CONGOPE-2015-009    NaN  \
1035      392170-CDC-CONGOPE-2015-014  1708541-CDC-CONGOPE-2015-014    NaN   
1173      392275-CDC-EASYNET-009-2015  1801313-CDC-EASYNET-009-2015    NaN   
2134       374115-CDC-GADMIS-010-2015   1461599-CDC-GADMIS-010-2015    NaN   
2137       384017-CDC-GADMIS-013-2015   1484908-CDC-GADMIS-013-2015    NaN   
...                               ...                           ...    ...   
1504511    2044037-RE-MIDUVI-002-2020    4416785-RE-MIDUVI-002-2020    NaN   
1746821  2124424-CTBS-SRIPCH-001-2021  4933881-CTBS-SRIPCH-001-2021    NaN   
1750619    2094076-RE-CS-ANDEC-001-21    4722794-RE-CS-ANDEC-001-21    NaN   
1860459     2124561-FONAG-RE-009-2022     5621002-FONAG-RE-009-2022    NaN   
1922094    2152051-MCO-BALAG-003-2022    5874765-MCO-BALAG-003-2022    NaN   

         description      status   contractPeriod_startDate   
1031             NaN  terminated  2015-09-11T00:00:00-05:00  \
1035             NaN  terminated  2015-12-15T00:00:00-05:00   
1173             NaN  terminated  2016-01-29T00:00:00-05:00   
2134             NaN  terminated  2015-07-28T00:00:00-05:00   
2137             NaN  terminated  2015-08-03T00:00:00-05:00   
...              ...         ...                        ...   
1504511          NaN  terminated  2020-10-01T00:00:00-05:00   
1746821          NaN  terminated  2022-09-26T00:00:00-05:00   
1750619          NaN  terminated  2021-07-01T00:00:00-05:00   
1860459          NaN  terminated  2022-06-16T00:00:00-05:00   
1922094          NaN      active  2022-10-07T00:00:00-05:00   

            contractPeriod_endDate  contractPeriod_maxExtentDate   
1031     2015-12-10T00:00:00-05:00                           NaN  \
1035     2015-12-16T00:00:00-05:00                           NaN   
1173     2016-02-01T00:00:00-05:00                           NaN   
2134     2015-08-27T00:00:00-05:00                           NaN   
2137     2015-08-08T00:00:00-05:00                           NaN   
...                            ...                           ...   
1504511  2021-01-19T00:00:00-05:00                           NaN   
1746821  2022-09-27T00:00:00-05:00                           NaN   
1750619  2021-12-31T00:00:00-05:00                           NaN   
1860459  2022-09-14T00:00:00-05:00                           NaN   
1922094  2022-10-17T00:00:00-05:00                           NaN   

         contractPeriod_durationInDays  amount currency   
1031                              90.0    0.01      USD  \
1035                               1.0    0.01      USD 

In [9]:
contracts[contracts['amount']==contracts['amount'].max()]

ocid   
1504852  ocds-5wno2w-RE-SENAE-005-2020-2656  \

                                              release_id   
1504852  RE-SENAE-005-2020-2656-2021-11-30T23:43:23.299Z  \

                                id                    awardID  title   
1504852  2054531-RE-SENAE-005-2020  4524968-RE-SENAE-005-2020    NaN  \

         description      status   contractPeriod_startDate   
1504852          NaN  terminated  2020-11-26T00:00:00-05:00  \

            contractPeriod_endDate  contractPeriod_maxExtentDate   
1504852  2021-05-27T00:00:00-05:00                           NaN  \

         contractPeriod_durationInDays        amount currency   
1504852                          182.0  8.978000e+12      USD  \

                        dateSigned   año  
1504852  2020-11-26T00:00:00-05:00  2020